src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Tree/Dependencies/DependencyTreeTelemetryService.cs

```diff
@@ -153,11 +157,83 @@ async Task<string> GetProjectIdAsync()
                 }
                 else
                 {
-                    return _telemetryService!.HashValue(_project.FullPath);
+                    return _telemetryService.HashValue(_project.FullPath);
                 }
             }
         }
 
+        public void ObserveSnapshot(DependenciesSnapshot dependenciesSnapshot)
+        {
+            _dependenciesSnapshot = dependenciesSnapshot;
+        }
+
+        public void Dispose()
+        {
+            if (_telemetryService != null && _dependenciesSnapshot != null && _projectId != null)
+            {
+                var data = new Dictionary<string, Dictionary<string, DependencyCount>>();
+                int totalDependencyCount = 0;
+                int unresolvedDependencyCount = 0;
+
+                // Scan the snapshot and tally dependencies
+                foreach ((TargetFramework targetFramework, TargetedDependenciesSnapshot targetedSnapshot) in _dependenciesSnapshot.DependenciesByTargetFramework)
+                {
+                    var countsByType = new Dictionary<string, DependencyCount>();
+
+                    data[targetFramework.ShortName] = countsByType;
+
+                    foreach (IDependency dependency in targetedSnapshot.Dependencies)
+                    {
+                        // Only include visible dependencies in telemetry counts
+                        if (!dependency.Visible)
+                        {
+                            continue;
+                        }
+
+                        if (!countsByType.TryGetValue(dependency.ProviderType, out DependencyCount counts))
+                        {
+                            counts = new DependencyCount();
+                        }
+
+                        countsByType[dependency.ProviderType] = counts.Add(dependency.Resolved);
+
+                        totalDependencyCount++;
+
+                        if (!dependency.Resolved)
+                        {
+                            unresolvedDependencyCount++;
+                        }
+                    }
+                }
+
+                _telemetryService.PostProperties(TelemetryEventName.ProjectUnloadDependencies, new (string, object)[]
+                {
+                    (TelemetryPropertyName.ProjectUnloadProject, _projectId),
+                    (TelemetryPropertyName.ProjectUnloadProjectAgeMillis, _projectLoadTime.ElapsedMilliseconds),
```

Comment:
What are you trying to gather from this? it has no real meaning to anything useful, other than the time that the dependency telemetry service was created to the time it disposed. We have existing events for determine how long the solution was open for, or project load times.

Reply:
The intention is to make it easy to filter out projects which were not alive for very long and for which all projects were unresolved, as they were likely to not have had long enough to complete and process the results of a design-time build. Unless it's trivial to join this event to other events from the same session, I'd rather keep this here for that reason.

Reply:
If you only report the data when we're up-to-date with builds as per below, that wouldn't happen. Using a time span as an indicator of whether the design-time builds results has been processed is not a good fit for telemetry, IMO.

Reply:
Agree, but it seemed like a pragmatic way to get indicative data without a clear way to get correct data. This approach would slightly overreport warnings, which is better than underreporting them. Let's discuss other approaches for more correct timing below.

The changes in #6183 would make it clearer when to publish this data, but we want a baseline before starting that work to see how much impact it will have.